### PR TITLE
azure_rm_networkinterface: Feature/dns servers

### PR DESCRIPTION
**Pull request by spmp**
##### SUMMARY
Add `dns_servers` option to `azure_rm_networkinterface`

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_networkinterface

##### ANSIBLE VERSION
```
2.7
```

### CODE REVIEW

File: test/integration/targets/azure_rm_networkinterface/tasks/main.yml [hunk 1]
```diff
@@ -1,29 +1,34 @@
 - name: Prepare random number
   set_fact:
     rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+    auth_source: auto
```
Comment (yuwzho) [inline test/integration/targets/azure_rm_networkinterface/tasks/main.yml]: no need to add auth_source

File: test/integration/targets/azure_rm_networkinterface/tasks/main.yml [hunk 2]
```diff
@@ -5,17 +5,17 @@
 
 - name: Create virtual network
   azure_rm_virtualnetwork:
-      resource_group: "{{ resource_group_secondary }}"
-      name: "tn{{ rpfx }}"
-      address_prefixes: "10.10.0.0/16"
+    resource_group: "{{ resource_group_secondary }}"
```
Comment (yuwzho) [inline test/integration/targets/azure_rm_networkinterface/tasks/main.yml]: Hi @spmp As we talked this morning, please make the test with 4 spaces

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 3]
```diff
@@ -512,6 +529,16 @@ def exec_module(self, **kwargs):
                         results.get('enable_accelerated_networking')))
                     changed = True
 
+                # We need to ensure that dns_servers are list like
+                dns_servers_res = results.get('dns_settings').get('dns_servers')
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]: Null check for dns_settings here?

**Review by yuwzho — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py (hunk 3 above)
Comment (spmp) [reply lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]: `dns_settings` should never be null as it comes from `network_client.network_interfaces.get()`
Certainly can test, but we could miss breaking API changes from Azure SDK if we do this...

Just for fun: The DNS settings does not conform to published API from a setting perspective BTW. I had an interesting conversation with the Azure Architect (or _an_ architect) about this.... turns out you cannot set anything except `dns_servers`, hence only this option. Well, you can set them, they are simply ignored.

**Review by yuwzho — APPROVED** [review]

**Comment by yuwzho** [comment]
shipit

**Comment by yungezz** [comment]
shipit

**Comment by yungezz** [comment]
HI @spmp could you pls resolve conflicts?

**Comment by Fred-sun** [comment]
@spmp Thanks for your contribution, Could you please resolve conflicting files?

**Comment by spmp** [comment]
@Fred-sun rebased

**Comment by Fred-sun** [comment]
@spmp Thanks for your contribution, Does this PR ready for review or not?

**Comment by spmp** [comment]
@Fred-sun yes this is ready for review

**Comment by Fred-sun** [comment]
@spmp Thanks for your update, I will push review. Thanks!

**Comment by Fred-sun** [comment]
@yuwzho Could you help approve those change? Thanks!

**Comment by yuwzho** [comment]
@jborean93 for a further review

**Comment by Fred-sun** [comment]
@jborean93 Could you help take a look this PR? Thanks!

**Comment by Fred-sun** [comment]
@jborean93 Could you help take a look this PR? Thanks!

**Comment by Fred-sun** [comment]
@jborean93 Could you help take a look this PR? Thanks!

**Comment by Fred-sun** [comment]
@jborean93 Could you help take a look this PR? we need push this PR to next step . Thanks!

**Review by nitzmahone — APPROVED** [review]

**Comment by spmp** [comment]
Cheers @nitzmahone

**Pull request merged**